### PR TITLE
[Bug-7832, Improvement] Add unit test for worker module, fix apache#7832. Optimize HttpUtils and HttpUtilsTest 1. Checking input parameters of getResponseContentString 2. Add unit test

### PR DESCRIPTION
**Pull request by springmonster**
## Purpose of the pull request

[Improvement] Optimize HttpUtils and HttpUtilsTest
1. Checking input parameters of getResponseContentString
2. Add unit test

## Brief change log

HttpsUtils.java and HttpsUtilsTest.java

## Verify this pull request

This pull request is already covered by existing tests

### CODE REVIEW

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java [hunk 1]
```diff
@@ -39,40 +39,41 @@
...
-	Assert.assertEquals("GitHub", jsonObject.path("name").asText());
-	result = HttpUtils.get("https://123.333.111.33/ccc");
-	Assert.assertNull(result);
+		// success
+		String result = HttpUtils.get("https://github.com/manifest.json");
+		Assert.assertNotNull(result);
+		ObjectNode jsonObject = JSONUtils.parseObject(result);
+		Assert.assertEquals("GitHub", jsonObject.path("name").asText());
+		result = HttpUtils.get("https://123.333.111.33/ccc");
+		Assert.assertNull(result);
```
Comment (zhuangchong) [inline dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java]: The code indentation format is wrong.

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java [hunk 2]
```diff
@@ -39,40 +39,41 @@
...
-	}
-
+		try {
+			String applicationUrl = hadoopUtils.getApplicationUrl("application_1542010131334_0029");
+			String responseContent;
+			responseContent = HttpUtils.get(applicationUrl);
+			Assert.assertNull(responseContent);
+		} catch (Exception e) {
+			logger.error(e.getMessage(), e);
+		}
```
Comment (zhuangchong) [inline dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java]: The code indentation format is wrong.

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java [hunk 3]
```diff
@@ -39,40 +39,41 @@
...
+		/** set timeout、request time、socket timeout */
+		RequestConfig requestConfig = RequestConfig.custom().setConnectTimeout(Constants.HTTP_CONNECT_TIMEOUT)
+			.setConnectionRequestTimeout(Constants.HTTP_CONNECTION_REQUEST_TIMEOUT)
+			.setSocketTimeout(Constants.SOCKET_TIMEOUT).setRedirectsEnabled(true).build();
+		httpget.setConfig(requestConfig);
+		String responseContent = HttpUtils.getResponseContentString(httpget, httpclient);
+		Assert.assertNotNull(responseContent);
+	
+		responseContent = HttpUtils.getResponseContentString(null, null);
+		Assert.assertNull(responseContent);
```
Comment (zhuangchong) [inline dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java]: The code indentation format is wrong.

**Review by zhuangchong — CHANGES_REQUESTED** [review]

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HttpUtils.java [hunk 4]
```diff
@@ -154,10 +154,10 @@ public static String getResponseContentString(HttpGet httpget, CloseableHttpClie
                     logger.warn("http entity is null");
                 }
             } else {
-                logger.error("http get:{} response status code is not 200!", response.getStatusLine().getStatusCode());
+                logger.error("http get: {} response status code is not 200!", response.getStatusLine().getStatusCode());
             }
-        } catch (IOException ioe) {
-            logger.error(ioe.getMessage(), ioe);
+        } catch (NullPointerException | IOException exception) {
+            logger.error(exception.getMessage(), exception);
```
Comment (zhuangchong) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HttpUtils.java]: I would recommend to verify the parameters before the method is processed.

**Review by zhuangchong — CHANGES_REQUESTED** [review]

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HttpUtils.java (hunk 4 above)
Comment (springmonster) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HttpUtils.java]: Hi @zhuangchong ，updated，please check again.
Thanks for your suggestion.

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java (hunk 3 above)
Comment (springmonster) [reply dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java]: Hi @zhuangchong ，updated，please check again.
Thanks for your suggestion.

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java (hunk 2 above)
Comment (springmonster) [reply dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java]: Hi @zhuangchong ，updated，please check again.
Thanks for your suggestion.

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java (hunk 1 above)
Comment (springmonster) [reply dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/HttpUtilsTest.java]: Hi @zhuangchong ，updated，please check again.
Thanks for your suggestion.

**Review by springmonster — COMMENTED** [review]
👍🏻

**Review by zhuangchong — APPROVED** [review]
+1

**Pull request merged**